test/e2e/operator.go

```diff
@@ -123,39 +109,27 @@ var _ = Describe("ARO Operator - Internet checking", func() {
 		Expect(err).NotTo(HaveOccurred())
 
 		By("waiting for the expected conditions to be set")
-		err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
+		Eventually(func(g Gomega, ctx context.Context) {
 			co, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, "cluster", metav1.GetOptions{})
-			if err != nil {
-				log.Warn(err)
-				return false, nil // swallow error
-			}
+			g.Expect(err).NotTo(HaveOccurred())
 
-			log.Debugf("ClusterStatus.Conditions %s", co.Status.Conditions)
-			return conditions.IsFalse(co.Status.Conditions, arov1alpha1.InternetReachableFromMaster) &&
-				conditions.IsFalse(co.Status.Conditions, arov1alpha1.InternetReachableFromWorker), nil
-		})
-		Expect(err).NotTo(HaveOccurred())
+			g.Expect(conditions.IsFalse(co.Status.Conditions, arov1alpha1.InternetReachableFromMaster)).To(BeTrue())
+			g.Expect(conditions.IsFalse(co.Status.Conditions, arov1alpha1.InternetReachableFromWorker)).To(BeTrue())
+		}).WithContext(ctx).Should(Succeed())
 	})
 })
 
 var _ = Describe("ARO Operator - Geneva Logging", func() {
 	It("must be repaired if DaemonSet deleted", func(ctx context.Context) {
-		mdsdReady := func() (bool, error) {
+		mdsdIsReady := func(g Gomega, ctx context.Context) {
 			done, err := ready.CheckDaemonSetIsReady(ctx, clients.Kubernetes.AppsV1().DaemonSets("openshift-azure-logging"), "mdsd")()
-			if err != nil {
-				log.Warn(err)
-			}
-			return done, nil // swallow error
+
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(done).To(BeTrue())
 		}
 
 		By("checking that mdsd DaemonSet is ready before the test")
-		err := wait.PollImmediate(30*time.Second, 15*time.Minute, mdsdReady)
```

Comment:
`Eventually` will be using default polling and wait intervals set here:

https://github.com/Azure/ARO-RP/blob/00f1604fd229bf5b4e92f96b83040741b5a99494/test/e2e/setup.go#L411-L412

They are lower than what we had, but we can increase themon per-`Eventually` basis if we need it.

But I don't think that any of these waits need to be 15 minutes: I believe these situations either recover quickly or do not recover at all.